ros_ws/src/teleoperation/src/joystick_controller.cpp

```diff
@@ -40,19 +55,12 @@ void JoystickController::joystickCallback(const sensor_msgs::Joy::ConstPtr& joys
         this->m_dms_publisher.publish(dms_message);
     }
 
-    // compute and publish angle and velocity
-    double steering_angle = joystick->axes[JOYSTICK_AXIS_STEERING] * -1.0f;
-    double velocity = (joystick->axes[JOYSTICK_AXIS_THROTTLE] - 1) * -0.5f;
-
-    this->publishDriveParameters(velocity, steering_angle);
+    // compute and publish the provided steering and velocity
+    float acceleration = m_joystick_map->getAcceleration(joystick) * ACCELERATION_SCALING_FACTOR;
+    float deceleration = m_joystick_map->getDeceleration(joystick) * DECELERATION_SCALING_FACTOR;
+    float steering_angle = m_joystick_map->getSteeringAxis(joystick) * STEERING_SCALING_FACTOR;
 
-    // Detect if the button was pressed since the last reading
-    bool invert_toggle_button = joystick->buttons[JOYSTICK_BUTTON_TOGGLE_INVERT_STEERING] == 1;
-    if (invert_toggle_button && !this->m_toggle_invert_steering_state)
-    {
-        this->m_invert_steering = !this->m_invert_steering;
-    }
-    this->m_toggle_invert_steering_state = invert_toggle_button;
+    this->publishDriveParameters(acceleration - deceleration, steering_angle);
```

Comment:
**MAY**
Before publishing the parameters a final sanity check might be useful like are the values for the `acceleration` in the expected range, is the difference between `acceleration` and `deceleration` in the expected range.